### PR TITLE
fix: provide proper error when POSTing to a missing page endpoint

### LINKED ISSUE #8696: POSTing to a page without actions should result in a 405
### Describe the bug

https://twitter.com/bradgarropy/status/1617733494490423298

If you POST to a page without an action, you should get a 405 response, but instead you get a 200. I could have sworn this already happened, but it doesn't — we only 405 in different circumstances

### Reproduction

https://stackblitz.com/edit/sveltejs-kit-template-default-cvxk7v?file=src%2Froutes%2F%2Bpage.svelte,svelte.config.js&terminal=dev

### Logs

_No response_

### System Info

```Shell
System:
    OS: macOS 12.5.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 442.55 MB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.17.0 - ~/.nvm/versions/node/v16.17.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.17.0/bin/yarn
    npm: 8.15.0 - ~/.nvm/versions/node/v16.17.0/bin/npm
  Browsers:
    Chrome: 109.0.5414.87
    Chrome Canary: 111.0.5553.0
    Firefox: 109.0
    Safari: 15.6.1
  npmPackages:
    @sveltejs/adapter-auto: ^1.0.0 => 1.0.2 
    @sveltejs/adapter-cloudflare: ^1.0.0 => 1.1.0 
    @sveltejs/adapter-cloudflare-workers: ^1.0.0 => 1.0.6 
    @sveltejs/adapter-netlify: ^1.0.0 => 1.0.5 
    @sveltejs/adapter-node: ^1.0.0 => 1.1.4 
    @sveltejs/adapter-static: ^1.0.0 => 1.0.5 
    @sveltejs/adapter-vercel: ^1.0.0 => 1.0.5 
    @sveltejs/kit: ^1.0.0 => 1.2.5 
    svelte: ^3.55.1 => 3.55.1 
    vite: ^4.0.4 => 4.0.4
```

### Severity

annoyance

### Additional Information

_No response_

### LINKED ISSUE #8713: Cryptic error when no route and using form enhance
### Describe the problem

When using the form `enhance` function a cryptic error occurs when there is no action associated with that route. I found this when I accidentally left a '+' sign off of a `page.server.js` page.

```
ReferenceError: error is not defined
    at Module.handle_action_json_request (/node_modules/@sveltejs/kit/src/runtime/server/page/actions.js:28:28)
    at Module.render_page (/node_modules/@sveltejs/kit/src/runtime/server/page/index.js:45:9)
    at async resolve (/node_modules/@sveltejs/kit/src/runtime/server/respond.js:362:17)
    at async Module.respond (/node_modules/@sveltejs/kit/src/runtime/server/respond.js:233:20)
    at async file:///Users/glen/tmp/my-app/node_modules/@sveltejs/kit/src/exports/vite/dev/index.js:484:22
```

Minimal repro is here: https://github.com/Glench/sveltekit-cryptic-enhance-error

### Describe the proposed solution

An error like `The form action 'default' at '/route' doesn't exist. Are you sure there's a '+page.server.js' file?`

### Alternatives considered

_No response_

### Importance

nice to have

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by gtm-nayan**
Closes #8696 
Closes #8713 

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

**Review by dummdidumm — APPROVED** [review]
Thank you!

File: packages/kit/src/runtime/server/page/actions.js [hunk 1]
```diff
@@ -69,23 +69,23 @@ export async function handle_action_json_request(event, options, server) {
 			});
 		}
 	} catch (e) {
-		const error = normalize_error(e);
+		const err = normalize_error(e);
```
Comment (dummdidumm) [inline packages/kit/src/runtime/server/page/actions.js:72]: FYI @bluwy , I think there's a Vite bug where it thinks that `error` is the function imported at the top, not the `const` declared here.

File: packages/kit/src/runtime/server/page/actions.js [hunk 2]
```diff
@@ -110,19 +110,19 @@ function action_json(data, init) {
 
 /**
  * @param {import('types').RequestEvent} event
- * @param {import('types').SSRNode} leaf_node
  */
-export function is_action_request(event, leaf_node) {
-	return leaf_node.server && event.request.method !== 'GET' && event.request.method !== 'HEAD';
+export function is_action_request(event) {
+	const { method } = event.request;
+	return method !== 'GET' && method !== 'HEAD';
```
Comment (Rich-Harris) [inline packages/kit/src/runtime/server/page/actions.js:116]: On my phone so viewing the code a few lines at a time and could be misunderstanding, but shouldn't this be checking that method is POST?

Comment (gtm-nayan) [reply packages/kit/src/runtime/server/page/actions.js:116]: Was confused by that as well, but didn't want to trigger any [unwanted changes](https://xkcd.com/1172/) so I just kept it as is.

PATCH, DELETE don't ever reach this point but other unknown methods might, do you want to tighten it down?

Comment (dummdidumm) [reply packages/kit/src/runtime/server/page/actions.js:116]: This is probably a reminiscent of old times, yes we can simplify that to just check for POST

File: packages/kit/src/runtime/server/page/actions.js (hunk 1 above)
Comment (bluwy) [reply packages/kit/src/runtime/server/page/actions.js:72]: 👍 looks like @gtm-nayan got it fixed at https://github.com/vitejs/vite/pull/11811

Comment (dummdidumm) [reply packages/kit/src/runtime/server/page/actions.js:72]: Oh nice 🎉

**Pull request merged**